packages/commands/src/hooks/use-command.js

```diff
@@ -0,0 +1,41 @@
+/**
+ * WordPress dependencies
+ */
+import { useEffect, useRef } from '@wordpress/element';
+import { useDispatch } from '@wordpress/data';
+
+/**
+ * Internal dependencies
+ */
+import { store as commandsStore } from '../store';
+
+/**
+ * Attach a command to the Global command menu.
+ *
+ * @param {import('../store/actions').WPCommandConfig} command command config.
+ */
+export default function useCommand( command ) {
+	const { registerCommand, unregisterCommand } = useDispatch( commandsStore );
+	const currentCallback = useRef( command.callback );
+	useEffect( () => {
+		currentCallback.current = command.callback;
+	}, [ command.callback ] );
+
+	useEffect( () => {
+		registerCommand( {
+			name: command.name,
+			group: command.group,
+			label: command.label,
+			callback: currentCallback.current,
```

Comment:
To avoid re-registering the command if the callback is an inline function. We only care about the callback at the moment it gets called, so a ref storing the latest version is sufficient.